modules/tensor_mechanics/include/materials/SingleVariableReturnMappingSolution.h

```diff
@@ -32,23 +40,25 @@ class SingleVariableReturnMappingSolution
    * @param scalar                 Inelastic strain increment magnitude being solved for
    * @param console                Console output
    */
-  void returnMappingSolve(const Real effective_trial_stress,
-                          Real & scalar,
+  void returnMappingSolve(const GenericReal<is_ad> & effective_trial_stress,
+                          GenericReal<is_ad> & scalar,
                           const ConsoleStream & console);
 
   /**
    * Compute the minimum permissible value of the scalar.  For some models, the magnitude
    * of this may be known.
    * @param effective_trial_stress Effective trial stress
    */
-  virtual Real minimumPermissibleValue(const Real & effective_trial_stress) const;
+  virtual GenericReal<is_ad>
+  minimumPermissibleValue(const GenericReal<is_ad> & effective_trial_stress) const;
 
   /**
    * Compute the maximum permissible value of the scalar.  For some models, the magnitude
    * of this may be known.
    * @param effective_trial_stress Effective trial stress
    */
-  virtual Real maximumPermissibleValue(const Real & effective_trial_stress) const;
+  virtual GenericReal<is_ad>
+  maximumPermissibleValue(const GenericReal<is_ad> & effective_trial_stress) const;
```

Comment:
should those min/max permissible values really be dual numbers in the AD case?

Reply:
I suppose it won;t hurt

Reply:
Yes, I pass a ADReal in bison for this, based on other ADReals.